src/ProjectSystem/Impl/Extensions/ProjectTreeExtensions.cs

```diff
@@ -63,6 +63,38 @@ public static class ProjectTreeExtensions {
             return Enumerable.Empty<string>();
         }
 
+        public static IEnumerable<string> GetAllFilePaths(this IEnumerable<IProjectTree> nodes) {
+            if (nodes != null) {
+                List<string> paths = new List<string>();
+                foreach (IProjectTree node in nodes) {
+                    if (node.IsFolder) {
+                        paths.AddRange(node.GetAllFilePaths());
+                    } else if (node.Children.Count > 0) {
+                        paths.AddRange(node.Children.GetAllFilePaths());
+                    } else if (!string.IsNullOrWhiteSpace(node.FilePath)) {
+                        paths.Add(node.FilePath);
+                    }
+                }
+                return paths;
+            }
+            return Enumerable.Empty<string>();
+        }
+
+        public static IEnumerable<string> GetAllFilePaths(this IProjectTree node) {
```

Comment:
simplified this to a single fucntion

---
In reply to: [80773143](https://github.com/Microsoft/RTVS/pull/2405#discussion_r80773143) [](ancestors = 80773143,80771759,80537483,80395305)